js/src/util/bn.ts

```diff
@@ -91,15 +91,25 @@ export function bigNumToNumber<T extends BN<BigNumArray>>(bn: T, scale?: number)
             number |= word * (BigInt(1) << BigInt(64 * i++));
         }
     }
-    if (typeof scale === 'number') {
-        const denominator = BigInt(Math.pow(10, scale));
+    if (typeof scale === 'number' && scale > 0) {
+        const denominator = BigInt('1' + '0'.repeat(scale));
```

Comment:
Can use `padEnd()` here:
```suggestion
        const denominator = BigInt('1'.padEnd(scale + 1, '0'));
```